doc/reference/devices_disk.md

```diff
@@ -70,6 +70,21 @@ VM `cloud-init`
 
       lxc config device add <instance_name> <device_name> disk source=cloud-init:config
 
+## Initial configuration for instance root disk devices
+
+Initial configuration allows setting specific configurations for the root disk devices of new instances.
+These settings are prefixed with `initial.` and are only applied when the instance is created.
+This allows creating instances that have unique configurations, independent of the default storage pool settings.
+
+For example, you can add an initial configuration for `block.filesystem` to an existing profile, and this will then
+take effect for each new instance you create using this profile:
+
+    lxc profile device set <profile_name> <device_name> initial.block.filesystem=btrfs
+
+You can also set an initial configuration directly when creating an instance. For example:
```

Comment:
Could you also add an example of enabling zfs block mode in this section as that is what the original request was for.

Thanks